parler_tts/modeling_parler_tts.py

```diff
@@ -244,7 +248,7 @@ def get_embedding(num_embeddings: int, embedding_dim: int):
     def forward(self, input_ids: torch.Tensor, past_key_values_length: int = 0):
         bsz, seq_len, _ = input_ids.size()
         # Create the position ids from the input token ids.
-        position_ids = (torch.arange(seq_len) + past_key_values_length).to(input_ids.device)
+        position_ids = (torch.arange(seq_len, device=input_ids.device) + past_key_values_length).to(input_ids.device)
```

Comment:
Nice catch.
Issue here is that there is an inconsistency in `past_key_values_length` type: can be either 0, so an int, and a tensor (code taken from [here](https://github.com/huggingface/transformers/blob/c1aa0edb48217f416f4bbe6e3a9db1500284513b/src/transformers/models/whisper/modeling_whisper.py#L1300C5-L1304C23)). This inconsistency is well handled by torch when doing `torch.arange(seq_len) + past_key_values_length`, yet when `past_key_values_length` is a tensor, not on cpu device, we need to make sure that `torch.arange` creates tensor on the same device. I did not use `past_key_values_length.device` to avoid the case where `past_key_values_length` is `int` 0, and used this trick instead.

Reply:
Agree that this is not very elegant, yet I don't see a more elegant way to do it. Maybe change the logic copied from [whisper's code ](https://github.com/huggingface/transformers/blob/c1aa0edb48217f416f4bbe6e3a9db1500284513b/src/transformers/models/whisper/modeling_whisper.py#L1300C5-L1304C23)that introduces this inconsistency ?